components/BarcodeScanner.vue

```diff
@@ -138,28 +155,46 @@ export default {
       cameraConstraints.zoom.step = step;
     }
 
+    async function emitScanResult({barcodeDetector, video}) {
+      const {signal} = abortController;
+      const barcodes = await detectBarcodes({barcodeDetector, video, signal});
+      const [result] = barcodes;
+      emit('result', {type: result.format, text: result.rawValue});
+    }
+
+    function startBarcodeDetection() {
+      // Get video element
```

Comment:
Style nit; we use lowercase for `//` comments w/no required punctuation. We use complete sentences for multiline `/* ... */` comments.

```suggestion
      // get video element
```